plugins/node/opentelemetry-instrumentation-mysql2/test/mysql.test.ts

```diff
@@ -156,33 +163,76 @@ describe('mysql2', () => {
     memoryExporter.reset();
     instrumentation.setConfig({});
     instrumentation.disable();
-    connection.end(() => {
+    connection.end(err => {
       pool.end(() => {
         if (isPoolClusterEndIgnoreCallback()) {
           poolCluster.end();
-          done();
+          done(err);
         } else {
           // PoolCluster.end types in the package are invalid
           // https://github.com/sidorares/node-mysql2/pull/1332
           (poolCluster as any).end(() => {
-            done();
+            done(err);
           });
         }
       });
     });
   });
 
+  describe('initialization', () => {
```

Comment:
Why the addition of this test? Did you hit some specific issue?

1. I think this is not testing what you think it is testing. 

With both `instrumentation` and `instrumentation2` being enabled: the first one wraps, then the second one unwrap and wraps. Then when `.disable()` is called for both, the first one unwraps, and the second one is a no-op.

When running just this "initialization" test and with some added logging to show the value of `isWrapped(mysqlTypes.Connection.prototype.query)` before and after each enable/disable call, it looks like this:

```
% RUN_MYSQL_TESTS=1 npm test

> @opentelemetry/instrumentation-mysql2@0.43.0 test
> nyc mocha 'test/**/*.test.ts'



  mysql2
    initialization
XXX instr2.enable() isWrapped? false
XXX instr2.enabled isWrapped? true
XXX instr.enable() isWrapped? true
XXX instr.enabled isWrapped? true
      ✔ should not wrap more than once
XXX instr.disable() isWrapped? true
XXX instr.disabled isWrapped? false
XXX instr2.disable() isWrapped? false
XXX instr2.disabled isWrapped? false


  1 passing (17ms)
```

My diff for that test run is here: https://gist.github.com/trentm/d85145848dce0f45b8c440a6f655a078


2. We are actually considering *dropping* `Instrumentation#disable()` functionality altogether. See https://github.com/open-telemetry/opentelemetry-js/issues/5154 if you are curious.

My preference would be to not add this test.

Reply:
Honestly, I added it to reach 100% functions and lines coverage and see the branches not covered.
At the moment the only branch not covered is this one https://github.com/macno/opentelemetry-js-contrib/blob/issue-1758-mysql2-instrumentations-values-in-db.statement/plugins/node/opentelemetry-instrumentation-mysql2/src/instrumentation.ts#L80 which I have no idea how to test..